SingularityService/src/main/java/com/hubspot/singularity/resources/DeployResource.java

```diff
@@ -264,6 +264,11 @@ public SingularityRequestParent deploy(
       // This can cause a conflict if run outside the lock, causing the pending deploy to be checked before deploy data is saved
       schedulerLock.runWithRequestLock(
         () -> {
+          deployManager.createDeployIfNotExists(
+            updatedRequest,
+            deployMarker,
+            validatedDeploy
+          );
```

Comment:
Why do this before the deploy-already-in-progress check? We would already save this data right afterwards in saveDeploy anyway

Reply:
nevermind, I see what this is doing now, pending deploy gets created slightly before the actual deploy object. without it